python/tvm/relax/transform/optimize_layout_transform.py

```diff
@@ -0,0 +1,147 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, unused-argument
+"""Relax Optimize Layout Transform pass."""
+from typing import Union
+import tvm
+from tvm import IRModule, relax
+from tvm.ir.transform import module_pass
+from tvm.relax.analysis import remove_all_unused
+from tvm.relax.expr_functor import mutator, PyExprMutator
+
+
+@mutator
+class OptimizeLayoutTranformMutator(PyExprMutator):
```

Comment:
Can we use the built-in pattern rewriter instead?

https://github.com/apache/tvm/blob/unity/python/tvm/relax/dpl/rewrite.py#L26-L28

Reply:
Thanks @masahi for sharing this. 
I had gone through this earlier but I wasn't able to find a rewriter that can work on a complete IRModule. Hence, I implemented my own class. 
Do you know of a function that can take IRModule and rewrite all the patterns of interest ? 
If there is no such function, I can work on it. Basically it'll be a wrapper on top of `rewrite_call` to iterate over the whole graph.

Reply:
Why do you need a module-based pass for this rewriting? I think our pattern rewriter works at the function level partly because that's where the actual rewriting happens.

Reply:
Alright. I understand the concern here.  I will rewrite the pass as a function pass.